### ISSUE TITLE
Dragging a parent tab to its child freezes the browser (a machine)

### ISSUE THREAD

**Issue by hawran** [piroor/treestyletab#2245]
## Short description
Dragging a parent tab to its child freezes the browser (and a machine) and the browser has to be killed.
(experienced on a Linux machine)
I'd say there's some cycling problem here.

## Steps to reproduce
1. Create a tab.
2. Create the second tab as a child of the first tab.
3. Drag the first tab to its child.

## Expected result
Either a drag operation is cancelled or the dragged tab becomes the child of its child.

## Actual result
The browser freezes. And the machine as well.
The browser has to be killed.

## Environment
 * Platform (OS): Xbuntu 16.04.6 LTS (Linux 4.15.0-47-generic)
 * Version of Firefox: 66.0.3 (Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) Gecko/20100101 Firefox/66.0)
 * Version (or revision) of Tree Style Tab: 3.0.8

**Comment by stapuft** [comment]
cannot re-create on windows 10 17763 with FF 66.0.3

**Comment by piroor** [comment]
I couldn't reproduce this constantly on my environment:

* TST 3.0.8
* Nightly 68.0a1 / Firefox 66.0.3
* Windows 10

However, a problem similar to this happened just a time. After Firefox is restarted, I couldn't reproduce the problem with ctrl-drag.

----

Basically, TST is designed to treat such a case as "impossible to drop". I think that the internal tree structure was broken and TST failed to detect "you are dragging a parent onto its descendant" on your (and maybe my) case. Anyway, we need to narrow down exact steps to reproduce the problem from scratch.

**Comment by irvinm** [comment]
@hawran, can you reproduce this in a clean profile with just TST?  (Or can you reproduce consistently?)

**Comment by hawran** [comment]
> @hawran, can you reproduce this in a clean profile with just TST? (Or can you reproduce consistently?)

Hi irvinm,
Sorry for the delay.
Yes, I've just reproduced it within a clean profile with just TST:
```
Application Basics
------------------
Name: Firefox
Version: 66.0.5
Build ID: 20190508090509
User Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) Gecko/20100101 Firefox/66.0
OS: Linux 4.15.0-50-generic
Multiprocess Windows: 1/1 Enabled by default
Web Content Processes: 4/8
Enterprise Policies: Inactive

Extensions
----------
Name: Tree Style Tab
Version: 3.0.11
Enabled: true
ID: treestyletab@piro.sakura.ne.jp
```
Just drag the parent tab (see the screenshot attached) to the child tab.

![Screenshot_2019-05-18_20-31-02](https://user-images.githubusercontent.com/25924718/57973665-cd776300-79ac-11e9-8641-6cbadb879fc8.png)

**Comment by piroor** [comment]
Thanks, I've confirmed this problem on my Ubuntu 16.04LTS environment.

**Comment by piroor** [comment]
Fixed with the commit 2a72cfa

**Comment by hawran** [comment]
Thank you,
haw
